### PR TITLE
stop encoding strings in attrs

### PR DESCRIPTION
**Pull request by untzag**
leave strings utf8

@wright-group/wrighttools-maintainers please double check my work---this is a detail oriented patch and I have probably forgotten things

### CODE REVIEW

File: WrightTools/data/_data.py [hunk 1]
```diff
@@ -742,7 +742,7 @@ def map_variable(self, variable, points, input_units='same', *, name=None, paren
         if self.ndim == 1:
 
             def interpolate(dataset, points):
-                function = scipy.interpolate.interp1d(variable[:], dataset[:])
+                function = scipy.interpolate.interp1d(variable[:], dataset[:], bounds_error=False)
```
Comment (ksunden) [inline WrightTools/data/_data.py]: Unrelated?

**Comment by ksunden** [comment]
Should we be calling `encode()` on natural name (in`_group.py`, `_collection.py`)

what about `axes, channel_names, variable_names` (In `_data.py`), these three are lists/ndarray intermediate representation.

`decode()` is currently called on low level h5py interface (probably needed still), when dealing with `'item_names'` attr in  `_group.py` and in the corresponding places to above question within `_data.py`
```
[kyle@argon WrightTools]$ grep -r encode\(\)
_group.py:        elif self.natural_name.encode() not in parent.attrs['item_names']:
_group.py:                                                   self.natural_name.encode())
data/_data.py:        self.attrs['axes'] = [a.identity.encode() for a in self._axes]
data/_data.py:        self.attrs['channel_names'] = np.append(self.attrs['channel_names'], name.encode())
data/_data.py:        self.attrs['variable_names'] = np.append(self.attrs['variable_names'], name.encode())
collection/_collection.py:                                                 collection.natural_name.encode())
collection/_collection.py:            natural_name = "".encode()
collection/_collection.py:            natural_name = data.natural_name.encode()
[kyle@argon WrightTools]$ grep -r decode\(\)
_dataset.py:        fullpath = parent.fullpath + h5py.h5i.get_name(id).decode()
_group.py:        return tuple(n.decode() for n in self.attrs['item_names'])
data/_variable.py:        self.attrs['name'] = h5py.h5i.get_name(self.id).decode().split('/')[-1]
data/_channel.py:        self.attrs['name'] = h5py.h5i.get_name(self.id).decode().split('/')[-1]
data/_data.py:            identifier = identifier.decode()
data/_data.py:        return tuple(s.decode() for s in self.attrs['channel_names'])
data/_data.py:        return tuple(s.decode() for s in self.attrs['variable_names'])
```

File: WrightTools/data/_data.py (hunk 1 above)
Comment (untzag) [reply WrightTools/data/_data.py]: yes will you let me sneak it in :kissing_heart:
